--- 0form.inc	2012-04-26 06:41:32.000000000 -0400
+++ form.inc	2012-08-28 22:13:22.172170100 -0400
@@ -2646,7 +2646,7 @@
  */
 function form_select_options($element, $choices = NULL) {
   if (!isset($choices)) {
-    $choices = $element['#options'];
+    if (is_null($choices = $element['#options'])) return ''; // accommodation for dynamic javascript dropdowns -- #options = NULL
   }
   // array_key_exists() accommodates the rare event where $element['#value'] is NULL.
   // isset() fails in this situation.
